src/Core/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommand.cs

```diff
@@ -0,0 +1,354 @@
+
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var newSeatCount = organization.SmSeats.HasValue
+            ? organization.SmSeats.Value + update.SeatAdjustment
+            : update.SeatAdjustment;
+
+
+        if (update.MaxAutoscaleSeats.HasValue && newSeatCount > update.MaxAutoscaleSeats.Value)
+        {
+            throw new BadRequestException("Cannot set max seat autoscaling below seat count.");
+        }
+
+        var newServiceAccountCount = organization.SmServiceAccounts.HasValue
+            ? organization.SmServiceAccounts + update.ServiceAccountsAdjustment
+            : update.ServiceAccountsAdjustment;
+
+        if (update.MaxAutoscaleServiceAccounts.HasValue && newServiceAccountCount > update.MaxAutoscaleServiceAccounts.Value)
+        {
+            throw new BadRequestException("Cannot set max service account autoscaling below service account count.");
+        }
+
+        if (update.SeatAdjustment != 0)
+        {
+            await AdjustSeatsAsync(organization, update.SeatAdjustment);
+        }
+
+        if (update.ServiceAccountsAdjustment != 0)
+        {
+            await AdjustServiceAccountsAsync(organization, update.ServiceAccountsAdjustment.GetValueOrDefault());
+        }
+
+        if (update.MaxAutoscaleSeats != organization.MaxAutoscaleSmSeats)
+        {
+            UpdateSeatsAutoscaling(organization, update.MaxAutoscaleSeats);
+        }
+
+        if (update.MaxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts)
+        {
+            UpdateServiceAccountAutoscaling(organization, update.MaxAutoscaleServiceAccounts);
+        }
+
+        await _organizationService.ReplaceAndUpdateCacheAsync(organization);
+
+        if (organization.SmSeats.HasValue && organization.MaxAutoscaleSmSeats.HasValue && organization.SmSeats == organization.MaxAutoscaleSmSeats)
+        {
+            await SendEmailAsync(organization, organization.MaxAutoscaleSmSeats.Value, "Seats");
+        }
+
+        if (organization.SmServiceAccounts.HasValue && organization.MaxAutoscaleSmServiceAccounts.HasValue && organization.SmServiceAccounts == organization.MaxAutoscaleSmServiceAccounts
+            && update.ServiceAccountsAdjustment > 0)
+        {
+            await SendEmailAsync(organization, organization.MaxAutoscaleSmServiceAccounts.Value, "Service Accounts");
+        }
+    }
+
+    private async Task SendEmailAsync(Organization organization, int MaxAutoscaleValue, string adjustingProduct)
+    {
+        try
+        {
+            var ownerEmails = (await _organizationUserRepository.GetManyByMinimumRoleAsync(organization.Id,
+                    OrganizationUserType.Owner))
+                .Where(u => u.AccessSecretsManager)
+                .Select(u => u.Email).Distinct();
+
+            await _mailService.SendOrganizationMaxSeatLimitReachedEmailAsync(organization, MaxAutoscaleValue, ownerEmails);
```

Comment:
You have the `adjustingProduct` parameter but we're not doing anything with it. I assume this email needs to be dynamic to specify PM seats vs. SM seats vs. SM service accounts?

Or it might be better to just create separate emails - for example, it has a link to the help center article which will be different for different products.

Please check with product what wording they want to use for these emails.

Reply:
Added the email implementation service account and seats. I have also reachout to Max to the contain but current using PM template and few change of word until Max confirms it.